### PR TITLE
Upgrade to Rails 7

### PR DESCRIPTION
**Pull request by jbirdjavi**

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -24,16 +24,8 @@ jobs:
...
-        uses: actions/setup-node@v3
-        with:
-          node-version: 16.20.0
-
-      - name: Install yarn dependencies and precompile assets
-        run: bundle exec rails assets:precompile
-
       - name: Rails tests
-        run: bundle exec rails test
+        run: CI=true bundle exec rails test
```
Comment (knutsenm) [inline .github/workflows/ruby.yml:28]: I don't think this is necessary; it's set by default for Actions. Repos that do Codecov rely on it without setting it explicitly, e.g. https://github.com/CruGlobal/summer_missions/blob/d8e411dbbadd891985be2f8a5d94b2b1cccc45b7/spec/spec_helper.rb#L10

**Review by knutsenm — CHANGES_REQUESTED** [review]

File: .github/workflows/ruby.yml (hunk 1 above)
Comment (knutsenm) [reply .github/workflows/ruby.yml:28]: Yep: https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

Comment (jbirdjavi) [reply .github/workflows/ruby.yml]: Sweet, thanks for verifying that!

**Review by knutsenm — APPROVED** [review]
🎉

**Pull request merged**